### PR TITLE
[SDK-3502] Add Android Smoke Tests

### PR DESCRIPTION
**Pull request by frederikprijck**
### Description

This PR adds an Android Smoke Test that tests the login and logout flow running in Circle.

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: auth0_flutter/example/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,9 +1,13 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
     package="com.auth0.auth0_flutter_example">
+
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```
Comment (Widcket) [inline auth0_flutter/example/android/app/src/main/AndroidManifest.xml:5]: Is this still needed now that we're not storing screenshots anymore?

File: auth0_flutter/example/android/app/src/debug/AndroidManifest.xml [hunk 2]
```diff
@@ -4,4 +4,6 @@
          to allow setting breakpoints, to provide hot reload, etc.
     -->
     <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (Widcket) [inline auth0_flutter/example/android/app/src/debug/AndroidManifest.xml:8]: Same, is this still needed?

File: .circleci/config.yml [hunk 3]
```diff
@@ -4,10 +4,56 @@ orbs:
...
+              restore-gradle-cache-prefix: v1a
+              post-emulator-launch-assemble-command: ""
+              post-emulator-wait-steps:
+                - run:
+                    name: Disable Chrome welcome prompt
+                    command: adb shell 'echo "chrome --disable-fre --no-default-browser-check --no-first-run" > /data/local/tmp/chrome-command-line'
+                - run:
+                    name: Enable ADB root
+                    command: adb root
+          - android/run-tests:
```
Comment (Widcket) [inline .circleci/config.yml:50]: As mentioned previously in Slack, please make sure to not run the smoke tests for PRs from forks.

Comment (stevehobbsdev) [reply .circleci/config.yml:50]: Do you have an example of what that config looks like?

Comment (Widcket) [reply .circleci/config.yml:50]: Yup https://github.com/auth0/nextjs-auth0/blob/main/.circleci/config.yml#L30

Comment (Widcket) [reply .circleci/config.yml:50]: I also included a screenshot with a red-squared example in yesterday's Slack message.

Comment (Widcket) [reply .circleci/config.yml:50]: There's also another example in this same file: https://github.com/auth0/auth0-flutter/blob/main/.circleci/config.yml#L123

**Comment by Widcket** [comment]
Please, make sure to update the instructions in https://github.com/auth0/auth0-flutter/blob/first-availability/DEVELOPMENT.md#integration-tests as well.

File: .circleci/config.yml (hunk 3 above)
Comment (frederikprijck) [reply .circleci/config.yml:50]: Will need to see how we address that here, as smoke tests isnt just a single step, it's an entire job.
Assumably, we need to do something comparable, just slightly different.

Will look into it, thanks.

File: auth0_flutter/example/android/app/src/androidTest/kotlin/com/auth0/auth0_flutter_example/SmokeTest.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
+package com.auth0.auth0_flutter_example
+
+import android.app.UiAutomation
+import android.content.Context
+import android.content.Intent
+import android.os.Environment
+import android.os.Environment.DIRECTORY_PICTURES
+import android.widget.Button
```
Comment (Widcket) [inline auth0_flutter/example/android/app/src/androidTest/kotlin/com/auth0/auth0_flutter_example/SmokeTest.kt:8]:
```suggestion
import android.content.Intent
import android.widget.Button
```
These are not needed anymore.

File: .circleci/config.yml [hunk 5]
```diff
@@ -89,6 +130,34 @@ jobs:
...
           upload_name: Auth0 Flutter Android
           flags: auth0_flutter_android
+  smoke_test_android:
+    parameters:
+        android_simulator:
+          type: string
+    executor:
+      name: android/android-machine
+      resource-class: xlarge
+      tag: 2021.10.1
```
Comment (Widcket) [inline .circleci/config.yml:140]:
```suggestion
      resource-class: xlarge
```
This is not necessary.

File: .circleci/config.yml [hunk 6]
```diff
@@ -4,10 +4,56 @@ orbs:
   macos: circleci/macos@2
   flutter: circleci/flutter@1
   codecov: codecov/codecov@3
+  local-android:
```
Comment (Widcket) [inline .circleci/config.yml:7]: Well done, nice way to break it down.

File: .circleci/config.yml [hunk 7]
```diff
@@ -89,6 +130,33 @@ jobs:
           file: ./auth0_flutter/example/build/auth0_flutter/reports/jacoco/debug/jacoco.xml
           upload_name: Auth0 Flutter Android
           flags: auth0_flutter_android
+  smoke_test_android:
+    parameters:
+        android_simulator:
+          type: string
+    executor:
+      name: android/android-machine
+      resource-class: xlarge
```
Comment (Widcket) [inline .circleci/config.yml:139]:
```suggestion
      resource-class: xlarge
      tag: 2022.06
```
Use latest tag.

File: .circleci/config.yml [hunk 8]
```diff
@@ -89,6 +130,34 @@ jobs:
...
+      tag: 2022.06
```
Comment (Widcket) [inline .circleci/config.yml:140]:
```suggestion
      tag: "2022.06"
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -89,6 +130,34 @@ jobs:
...
           upload_name: Auth0 Flutter Android
           flags: auth0_flutter_android
+  smoke_test_android:
+    parameters:
+        android_simulator:
+          type: string
+    executor:
+      name: android/android-machine
+      resource-class: xlarge
+      tag: "2022.06"
```
Comment (Widcket) [inline .circleci/config.yml:140]:
```suggestion
      tag: "2022.06.1"
```
Image alias from https://circleci.com/developer/images/image/cimg/android#image-tags does not seem to work.

**Review by Widcket — APPROVED** [review]

**Pull request merged**